update analyze concurrency for statistics (#15286)
pingcap/docs: statistics.md

> When you run the `ANALYZE` statement, you can adjust the concurrency using the following parameters, to control its effect on the system.

Reviewer: https://github.com/pingcap/docs-cn/pull/15389/files#r1391999220

The relationships of the relevant system variables are shown below:

![analyze_concurrency](/media/analyze_concurrency.png)

`tidb_build_stats_concurrency`, `tidb_build_sampling_stats_concurrency` and `tidb_analyze_partition_concurrency` are actually in a upstream-downstream relationship, as shown in the preceding diagram. The actual total concurrency is: `tidb_build_stats_concurrency` * (`tidb_build_sampling_stats_concurrency` + `tidb_analyze_partition_concurrency`). When changing these parameters, you need to consider the values of these three parameters at the same time. It is recommended to adjust `tidb_analyze_partition_concurrency`, `tidb_build_sampling_stats_concurrency`, `tidb_build_stats_concurrency` one by one, and observe the impact on the system. The larger the values of these three parameters, the greater the resource overhead on the system.

#### `tidb_build_stats_concurrency`

> When you run the `ANALYZE` statement, the task is divided into multiple small tasks. Each task only works on statistics of one column or index. You can use the `tidb_build_stats_concurrency` parameter to control the number of simultaneous small tasks. The default value is `2`. The default value is `4` for v7.4.0 and earlier versions.

Reviewer: https://github.com/pingcap/docs-cn/pull/15389/files#r1392010261

#### `tidb_build_sampling_stats_concurrency`

When analyzing ordinary columns, you can use [`tidb_build_sampling_stats_concurrency`](/system-variables.md#tidb_build_sampling_stats_concurrency-new-in-v750) to control the concurrency of executing sampling tasks. The default value is `2`.

#### `tidb_analyze_partition_concurrency`

> When running the `ANALYZE` statement, you can use [`tidb_analyze_partition_concurrency`](/system-variables.md#tidb_analyze_partition_concurrency) to control the concurrency of task writing. The default value is `2`. The default value is `1` for v7.4.0 and earlier versions.

Reviewer: https://github.com/pingcap/docs-cn/pull/15389/files#r1392024365

#### `tidb_distsql_scan_concurrency`

> When you analyze regular columns, you can use the [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) parameter to control the number of Region to be read at one time. The default value is `15`. Meanwhile, it will also affect query performance, please adjust carefully.

Reviewer 2: ```suggestion
When you analyze regular columns, you can use the [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) parameter to control the number of Region to be read at one time. The default value is `15`. Meanwhile, it will also affect query performance. Adjust the value carefully.
```

> When you analyze regular columns, you can use the [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) parameter to control the number of Regions to be read at one time. The default value is `15`. Note that changing the value will affect query performance. Adjust the value carefully.

Reviewer 2: ```suggestion
When you analyze regular columns, you can use the [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) parameter to control the number of Regions to be read at one time. The default value is `15`. Note that changing the value will affect query performance. You need to adjust the value carefully.
```

> When you analyze regular columns, you can use the [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) parameter to control the number of Regions to be read at one time. The default value is `15`. Note that changing the value will affect query performance. Adjust the value carefully.

Reviewer: ```suggestion
When you analyze regular columns, you can use the [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) variable to control the number of Regions to be read at one time. The default value is `15`. Note that changing the value will affect query performance. Adjust the value carefully.
```

#### `tidb_index_serial_scan_concurrency`

> When you analyze index columns, you can use the [`tidb_index_serial_scan_concurrency`](/system-variables.md#tidb_index_serial_scan_concurrency) parameter to control the number of Region to be read at one time. The default value is `1`. Meanwhile, it will also affect query performance, please adjust carefully.

Reviewer 2: ```suggestion
When you analyze index columns, you can use the [`tidb_index_serial_scan_concurrency`](/system-variables.md#tidb_index_serial_scan_concurrency) parameter to control the number of Regions to be read at one time. The default value is `1`. Meanwhile, it will also affect query performance. Adjust the value carefully.
```

> When you analyze index columns, you can use the [`tidb_index_serial_scan_concurrency`](/system-variables.md#tidb_index_serial_scan_concurrency) parameter to control the number of Regions to be read at one time. The default value is `1`. Note that changing the value will affect query performance. Adjust the value carefully.

Reviewer 2: ```suggestion
When you analyze index columns, you can use the [`tidb_index_serial_scan_concurrency`](/system-variables.md#tidb_index_serial_scan_concurrency) parameter to control the number of Regions to be read at one time. The default value is `1`. Note that changing the value will affect query performance. You need to adjust the value carefully.
```

> When you analyze index columns, you can use the [`tidb_index_serial_scan_concurrency`](/system-variables.md#tidb_index_serial_scan_concurrency) parameter to control the number of Regions to be read at one time. The default value is `1`. Note that changing the value will affect query performance. Adjust the value carefully.

Reviewer: ```suggestion
When you analyze index columns, you can use the [`tidb_index_serial_scan_concurrency`](/system-variables.md#tidb_index_serial_scan_concurrency) variable to control the number of Regions to be read at one time. The default value is `1`. Note that changing the value will affect query performance. Adjust the value carefully.
```

When you run the `ANALYZE` statement, you can adjust the concurrency using the following parameters, to control its effect on the system.

> > **Note:**
> >  
> > ![analyze_concurrency](/media/analyze_concurrency.png)
> >
> > `tidb_build_stats_concurrency`, `tidb_build_sampling_stats_concurrency` and `tidb_analyze_partition_concurrency` are actually in a upstream-downstream relationship, as shown in the diagram. The actual total concurrency is: `tidb_build_stats_concurrency` * (`tidb_build_sampling_stats_concurrency` + `tidb_analyze_partition_concurrency`). When changing these parameters, you need to consider the values of these three parameters at the same time. It is suggested to adjust `tidb_analyze_partition_concurrency`, `tidb_build_sampling_stats_concurrency`, `tidb_build_stats_concurrency` one by one, and observe the impact on the system. The larger the values of these three parameters, the greater the resource overhead on the system.

Reviewer 2: ```suggestion
The relationships of the relevant system variables are shown below:

![analyze_concurrency](/media/analyze_concurrency.png)

`tidb_build_stats_concurrency`, `tidb_build_sampling_stats_concurrency` and `tidb_analyze_partition_concurrency` are actually in a upstream-downstream relationship, as shown in the preceding diagram. The actual total concurrency is: `tidb_build_stats_concurrency` * (`tidb_build_sampling_stats_concurrency` + `tidb_analyze_partition_concurrency`). When changing these parameters, you need to consider the values of these three parameters at the same time. It is recommended to adjust `tidb_analyze_partition_concurrency`, `tidb_build_sampling_stats_concurrency`, `tidb_build_stats_concurrency` one by one, and observe the impact on the system. The larger the values of these three parameters, the greater the resource overhead on the system.
```

#### `tidb_build_stats_concurrency`

> Currently, when you run the `ANALYZE` statement, the task is divided into multiple small tasks. Each task only works on one column or index. You can use the `tidb_build_stats_concurrency` parameter to control the number of simultaneous tasks. The default value is `2`. The default value is `4` for v7.4.0 and earlier versions.

Reviewer 2: ```suggestion
When you run the `ANALYZE` statement, the task is divided into multiple small tasks. Each task only works on statistics of one column or index. You can use the `tidb_build_stats_concurrency` parameter to control the number of simultaneous small tasks. The default value is `2`. The default value is `4` for v7.4.0 and earlier versions.
```

#### `tidb_build_stats_concurrency`

Currently, when you run the `ANALYZE` statement, the task is divided into multiple small tasks. Each task only works on one column or index. You can use the `tidb_build_stats_concurrency` parameter to control the number of simultaneous tasks. The default value is `2`.

> #### `tidb_build_sampling_stats_concurrency`
> When performing the task of analyzing ordinary columns, `tidb_build_sampling_stats_concurrency`` can be used to control the concurrency of executing sampling tasks, and its default value is 2.
> 
> ### `tidb_analyze_partition_concurrency`
> When performing analyze, `tidb_analyze_partition_concurrency`` can be used to control the concurrency of task writing, and its default value is 2.

Reviewer 2: ```suggestion
#### `tidb_build_sampling_stats_concurrency`

When analyzing ordinary columns, you can use [`tidb_build_sampling_stats_concurrency`](/system-variables.md#tidb_build_sampling_stats_concurrency-new-in-v750) to control the concurrency of executing sampling tasks. The default value is `2`.

#### `tidb_analyze_partition_concurrency`

When running the `ANALYZE` statement, you can use [`tidb_analyze_partition_concurrency`](/system-variables.md#tidb_analyze_partition_concurrency) to control the concurrency of task writing. The default value is `2`.
```

#### `tidb_build_stats_concurrency`

> Currently, when you run the `ANALYZE` statement, the task is divided into multiple small tasks. Each task only works on one column or index. You can use the `tidb_build_stats_concurrency` parameter to control the number of simultaneous tasks. The default value is `2`.

Reviewer 2: ```suggestion
Currently, when you run the `ANALYZE` statement, the task is divided into multiple small tasks. Each task only works on one column or index. You can use the `tidb_build_stats_concurrency` parameter to control the number of simultaneous tasks. The default value is `2`. The default value is `4` for v7.4.0 and earlier versions.
```

#### `tidb_build_sampling_stats_concurrency`

When analyzing ordinary columns, you can use [`tidb_build_sampling_stats_concurrency`](/system-variables.md#tidb_build_sampling_stats_concurrency-new-in-v750) to control the concurrency of executing sampling tasks. The default value is `2`.

#### `tidb_analyze_partition_concurrency`

> When running the `ANALYZE` statement, you can use [`tidb_analyze_partition_concurrency`](/system-variables.md#tidb_analyze_partition_concurrency) to control the concurrency of task writing. The default value is `2`.

Reviewer 2: ```suggestion
When running the `ANALYZE` statement, you can use [`tidb_analyze_partition_concurrency`](/system-variables.md#tidb_analyze_partition_concurrency) to control the concurrency of task writing. The default value is `2`. The default value is `1` for v7.4.0 and earlier versions.
```

When you analyze index columns, you can use the `tidb_index_serial_scan_concurrency` parameter to control the number of Region to be read at one time. The default value is `1`.

> > **Note:**
> >
> > `tidb_build_stats_concurrency`, `tidb_build_sampling_stats_concurrency` and `tidb_analyze_partition_concurrency` are actually in a upstream-downstream relationship.
> > When changing these parameters, you need to consider the values of these three parameters at the same time. It is suggested to adjust
> > `tidb_analyze_partition_concurrency`, `tidb_build_sampling_stats_concurrency`, `tidb_build_stats_concurrency` one by one, and observe the impact on the system. The
> > larger the values of these three parameters, the greater the resource overhead on the system.

Reviewer 2: ```suggestion
> **Note:**
>
> `tidb_build_stats_concurrency`, `tidb_build_sampling_stats_concurrency` and `tidb_analyze_partition_concurrency` are actually in a upstream-downstream relationship. When changing these parameters, you need to consider the values of these three parameters at the same time. It is suggested to adjust `tidb_analyze_partition_concurrency`, `tidb_build_sampling_stats_concurrency`, `tidb_build_stats_concurrency` one by one, and observe the impact on the system. The larger the values of these three parameters, the greater the resource overhead on the system.
```
